lib/ts/controllers/s-modal.ts

```diff
@@ -0,0 +1,222 @@
+namespace Stacks {
+
+    export class ModalController extends Stacks.StacksController {
+        static targets = ["modal"];
+
+        private modalTarget!: HTMLElement;
+
+        private _boundClickFn!: any;
+        private _boundKeypressFn!: any;
+
+        private returnElement!: HTMLElement;
+
+        private _boundTabTrap!: any;
+
+        connect () {
+            this._validate();
+        }
+
+        /**
+         * Disconnects all added event listeners on controller disconnect
+         */
+        disconnect() {
+            this._unbindDocumentEvents();
+        };
+
+        /**
+         * Toggles the visibility of the modal
+         */
+        toggle () {
+            this._toggle();
+        }
+
+        /**
+         * Shows the modal
+         */
+        show () {
+            this._toggle(true);
+        }
+
+        /**
+         * Hides the modal
+         */
+        hide () {
+            this._toggle(false);
+        }
+
+        /**
+         * Validates the modal settings and attempts to set necessary internal variables
+         */
+        _validate() {
+            // check for returnElement support
+            var returnElementSelector = this.data.get("return-element");
+            if (returnElementSelector) {
+                this.returnElement = <HTMLElement>document.querySelector(returnElementSelector);
+
+                if (!this.returnElement) {
+                    throw "Unable to find element by return-element selector: " + returnElementSelector;;
```

Comment:
```suggestion
                    throw "Unable to find element by return-element selector: " + returnElementSelector;
```

Just a double semicolon.